### PR TITLE
Fix a problem where env vars do not have the right ordinal

### PR DESCRIPTION
**Pull request by dmlloyd**

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Comment by dmlloyd** [comment]
Could the failure be related? :worried:

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java [hunk 1]
```diff
@@ -88,6 +88,16 @@ public static void addSourceProvider(SmallRyeConfigBuilder builder, ConfigSource
...
         static final Pattern REP_PATTERN = Pattern.compile("[^a-zA-Z0-9_]");
 
+        private final int ordinal;
+
+        EnvConfigSource(final int ordinal) {
+            this.ordinal = ordinal;
+        }
+
+        public int getOrdinal() {
+            return ordinal;
```
[the remark this answers is not in the document]
Comment (dmlloyd) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java]: Either way works for me.  There's only one instance of the class so there's no significant space difference either way.

[the remark this answers is not in the document]
Comment (gastaldi) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java]: Unless you want to increase your LoC report, I'd prefer to return 300 directly without changing the constructor.  :)

[the remark this answers is not in the document]
Comment (dmlloyd) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java]: Ah, in this case it's important to make it be explicit to support #3670.

**Comment by gsmet** [comment]
Trying another round of CI with a rebase...

**Pull request merged**